### PR TITLE
Update Intel simd.conf

### PR DESCRIPTION
**Pull request by thiagomacieira**
This renames all "amx" entries to include a dash, as required by GCC.

It also updates the list of features to match Intel Instruction Set Extensions Manual release 048:
* amx-complex, for Granite Rapids (GNR)

I'm also adding the official names for:
* Sapphire Rapids - it's already on the Ark[1]
* Emerald Rapids - named in the DCAI Investor Day 2023[2]

Plus the following cores:
* Raptor Cove (RPC) - named in the Intel Technology Tour[3] and in the perfmon counters[4]

And codenames for the following processors:
* Arrow Lake
* Lunar Lake
* Clearwater Forest - named in the DCAI Investor Day 2023[2]

The two client processors (the "Lakes") were named in the Investor Day
2022[5], but I missed that announcement. The cores that they are marked
to use are definitely incorrect, but the actual names are not known yet.

[1] https://ark.intel.com/content/www/us/en/ark/products/series/228622/4th-generation-intel-xeon-scalable-processors.html
[2] https://www.intc.com/news-events/ir-calendar/detail/20230329-data-center-and-ai-investor-webinar / https://www.youtube.com/watch?v=z1Cyq6EdNSg
[3] https://download.intel.com/newsroom/2022/2022innovation/13th-Gen-Intel-Core-Desktop-Media-presentation.pdf
[4] https://perfmon-events.intel.com/ahybrid.html
[5] https://www.intel.com/content/www/us/en/newsroom/news/intel-technology-roadmaps-milestones.html#gs.yj89rf

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>

### CODE REVIEW

File: framework/simd.conf [hunk 1]
```diff
@@ -169,13 +170,17 @@ arch=CNL    	PLC
...
 arch=ADL	GRT
-#arch=RPL	GRT
+arch=RPL	GRT
 arch=MTL	CMT
+arch=ARL	CMT
+arch=LNL	CMT
 arch=ICX	SNC	pconfig
-arch=SPR	GLC	pconfig,amxtile,amxbf16,amxint8
-arch=GNR	GLC	pconfig,amxtile,amxbf16,amxint8,amxfp16,prefetchiti
+arch=SPR	GLC	pconfig,amx-tile,amx-bf16,amxint8
```
Comment (busykai) [inline framework/simd.conf:178]: amx-int8

File: framework/simd.conf [hunk 2]
```diff
@@ -169,13 +170,17 @@ arch=CNL    	PLC
...
+arch=EMR	SPR
+arch=GNR	GLC	pconfig,amx-tile,amx-bf16,amxint8,amx-fp16,amx-complex
```
Comment (busykai) [inline framework/simd.conf:180]: amx-int8

Comment (thiagomacieira) [reply framework/simd.conf:180]: Done

File: framework/simd.conf (hunk 1 above)
Comment (thiagomacieira) [reply framework/simd.conf:178]: Done

**Review by busykai — APPROVED** [review]

**Pull request merged**